src/QueryRenderer.ts

```diff
@@ -129,7 +134,7 @@ class QueryRenderChild extends MarkdownRenderChild {
 
     private async render({ tasks, state }: { tasks: Task[]; state: State }) {
         console.debug(
-            `Render ${this.queryType} called for ${tasks.length} tasks, state: ${state}`,
+            `Render ${this.queryType} called for a block in active file "${this.sourcePath}", to select from ${tasks.length} tasks: plugin state: ${state}`,
```

Comment:
**BUG** The file name in this message does not get updated if the user renames the file containing the task block during the session.

Reply:
If the file with the tasks block is closed and then re-opened, the filename is correctly updated.

Reply:
I'm not sure I understand the question, but I'll try and explain more clearly.

This comment does not affect anything in the display of tasks blocks in Live Preview and Reading mode. They all handle renames fine.

I wanted to include the name of the markdown file in the 'Render...' messages, so if there is a tasks query that displays hundreds of tasks that the user didn't know was open, such as if it was in a pinned file, they can decide if it is worth keeping open all the time.

In order to have access to the file name for these debug log messages, I passed it in to the class's constructor.

It seems that the QueryRenderChild objects get created when the markdown file containing the tasks block is opened. So on the rare occasions where someone renames an open markdown file that has a tasks block in, the debug log messages for that task block will keep reporting the original file name, until the file is closed and re-opened.

My feeling is that this is a cosmetic issue, not likely to affect many people. Whereas the slow editing is affecting many.

So my inclination is to not delay the upcoming release to fix this out-of-date-file-name-on-debug-log-messages-after-file-renamed issue.

Reply:
Thanks. The issue with handling at the point of renaming is that it would have to read the file to check whether there are any tasks blocks inside, and I'm inclined not to spend time on that now, to fix a rare debug log issue.

I've logged the problem in #899